### ISSUE TITLE
Fix invalid json for max warnings option

### ISSUE THREAD

**Issue by jsit** [stylelint/stylelint#5265]
> Clearly describe the bug

When using both the `--max-warnings` and `--formatter json` flags, the output is not valid json because there is a string at the end starting with `Max warnings exceeded:`

> Which version of stylelint are you using?

13.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`stylelint --formatter json --max-warnings 0 **/*.scss`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

Output should be valid json

**Comment by jeddy3** [comment]
@jsit Thanks for the report.

I can reproduce this locally.

Config:

```json
{
  "defaultSeverity": "warning",
  "rules": {
    "block-no-empty": true
  }
}
```

File:

```css
a {
}
```

Console:

```
npx stylelint "empty.css" --formatter json --max-warnings 0
```

Gives:

```json
[{"source":"/Users/jeddy3/Projects/stylelint-test/sources/css/empty.css","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":false,"warnings":[{"line":1,"column":3,"rule":"block-no-empty","severity":"warning","text":"Unexpected empty block (block-no-empty)"}]}]Max warnings exceeded: 1 found. 0 allowed
```

Offhand, I can't think where this may be coming from so a bit of digging may be needed.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.
